client_remote.go

```diff
@@ -43,14 +38,19 @@ type remoteSession struct {
 	// the parent session (ie the session running the listing)
 	parentSession ssh.Session
 
-	stdin io.Reader
+	stdin   io.Reader
+	cleanup func()
 }
 
 func (s *remoteSession) SetStdin(r io.Reader)  {}
 func (s *remoteSession) SetStdout(w io.Writer) {}
 func (s *remoteSession) SetStderr(w io.Writer) {}
 
 func (s *remoteSession) Run() error {
+	if s.cleanup != nil {
+		s.cleanup()
+		defer s.cleanup()
```

Comment:
A bit confused by this one: we directly `cleanup` and call it once more with `defer`? Is that really intentional?

Reply:
it is, not really sure its needed, but wouldn't hurt... the idea is to exhausts the readers before connecting and after disconnecting.